include/deal.II/particles/particle_handler.h

```diff
@@ -0,0 +1,480 @@
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_particles_particle_handler_h
+#define dealii_particles_particle_handler_h
+
+#include <deal.II/particles/particle.h>
+#include <deal.II/particles/particle_iterator.h>
+#include <deal.II/particles/property_pool.h>
+
+#include <deal.II/grid/tria.h>
+#include <deal.II/base/subscriptor.h>
+#include <deal.II/base/mpi.h>
+#include <deal.II/base/array_view.h>
+#include <deal.II/base/smartpointer.h>
+#include <deal.II/fe/mapping.h>
+
+#include <boost/serialization/map.hpp>
+#include <boost/range/iterator_range.hpp>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Particles
+{
+  /**
+   * This class manages the storage and handling of particles. It provides
+   * the data structures necessary to store particles efficiently, accessor
+   * functions to iterate over particles and find particles, and algorithms
+   * to distribute particles in parallel domains.
+   *
+   * @ingroup Particle
+   */
+  template <int dim, int spacedim=dim>
+  class ParticleHandler: public Subscriptor
+  {
+    public:
+      /**
+       * A type that can be used to iterate over all particles in the domain.
+       */
+      typedef ParticleIterator<dim,spacedim> particle_iterator;
+
+      /**
+       * A type that represents a range of particles.
+       */
+      typedef boost::iterator_range<particle_iterator> particle_iterator_range;
+
+      /**
+       * Default constructor.
+       */
+      ParticleHandler();
+
+      /**
+       * Constructor that initializes the particle handler with respect to
+       * a given triangulation and MPI communicator. Pointers to the
+       * triangulation and the communicator are stored inside of the particle
+       *
+       */
+      ParticleHandler(const parallel::distributed::Triangulation<dim,spacedim> &tria,
+                      const Mapping<dim,spacedim> &mapping,
+                      const MPI_Comm mpi_communicator,
+                      const unsigned int n_properties = 0);
+
+      /**
+       * Destructor.
+       */
+      ~ParticleHandler();
+
+      /**
+       * Initialize the particle handler. This function does not clear the
+       * internal data structures, it just sets the connections to the
+       * MPI communicator and the triangulation. This is useful after de-
+       * serialization of a particle handler.
+       */
+      void initialize(const parallel::distributed::Triangulation<dim,spacedim> &tria,
+                      const Mapping<dim,spacedim> &mapping,
```

Comment:
maybe add some documentation in in the class or here in `initialize` to explain what `mappings` is for.